lib/que/sql.rb

```diff
@@ -84,6 +84,55 @@ module Que
       LIMIT 1
     },
 
+    # instead of keeping this worker in a exclusivee queue mode let it take work
+    # from defined secondary_queues with work availble.
+    queue_permissive_lock_job: %{
+      WITH RECURSIVE jobs AS (
+        SELECT (j).*, pg_try_advisory_lock((j).job_id) AS locked
+        FROM (
+          SELECT j
+          FROM que_jobs AS j
+          WHERE job_id >= $2
+          AND (queue = $1::text OR queue = ANY($3::text[]))
+          AND run_at <= now()
+          AND retryable = true
+          ORDER BY
+            CASE WHEN queue = $1::text THEN 0 ELSE 1 END,
```

Comment:
```suggestion
            array_position($3::text[], queue),
```
If we have the "main" queue as the first element in the array we could have the order of the queue matter.

Reply:
Yeah thats nice, I've changed it over to use arrays which makes things a little simpler overall